src/scroll-timeline-base.js

```diff
@@ -114,9 +114,43 @@ export function calculateMaxScrollOffset(source, orientation) {
     return source.scrollWidth - source.clientWidth;
 }
 
+/**
+ * Calculates the maximum pixel value to use
+ * @param cssValue {CSSUnitValue}
+ * @param source {DOMElement}
+ * @param orientation {String}
+ * @returns {number}
+ */
+function calculateMaxValue(cssValue, source, orientation) {
+  if (cssValue instanceof CSSUnitValue) {
+    switch (cssValue.unit) {
+      case 'vw':
+        return Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+      case 'vh':
+        return Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);
+      case 'vmin':
+        return Math.min(
+          Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0),
+          Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0),
+        );
+      case 'vmax':
+        return Math.max(
+          Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0),
+          Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0),
+        );
+      default:
+        // NOOP
+    }
+  }
+
+  return orientation === "vertical"
```

Comment:
Orientation is resolved from block or inline to horizontal or vertical value within calculateScrollOffset without regard for the writing mode.  It is not being used elsewhere in the function. Suggest we move the logical to physical orientation mapping  to the top of this function (including the TODO unless fixed in the same PR).  directionAwareScrollOffset has the rules for mapping  block and inline.